File: exercicios/1914/1914.py
qt = int(input())

if qt >= 1 and qt <= 100:
    for i in range(qt):
        entrada = input().split()
        j1 = entrada[0]
        op1 = entrada[1]
        j2 = entrada[2]
        op2 = entrada[3]

        vl = input().split()
        v1 = int(vl[0])
        v2 = int(vl[1])
        if v1 >= 1 and v1 <= 1000000000 and v2 >= 1 and v2 <= 1000000000:
            
            if (v1 + v2) % 2 == 0:
                if op1 == 'PAR':
                    print(j1)
                else:
                    print(j2)
            if (v1 + v2) % 2 != 0:
                if op1 == 'IMPAR':
                    print(j1)
                else:
                    print(j2)

File: exercicios/R/1914/1914.r
input = file('stdin','r')

n = as.integer(readLines(input,n=1))

for(i in 1:n){
    line = strsplit(readLines(input,n=1), " ")

    p1 = line[[1]][1]
    op1 = line[[1]][2]

    p2 = line[[1]][3]
    op2 = line[[1]][4]

    valor = strsplit(readLines(input,n=1), " ")
    v1 = as.integer(valor[[1]][1])
    v2 = as.integer(valor[[1]][2])

    if((v1+v2)%%2==0){
        if(op1 == "PAR"){write(p1,'')}else{write(p2,'')}
    }
    if((v1+v2)%%2!=0){
        if(op1 == "IMPAR"){write(p1,'')}else{write(p2,'')}
    }
}